core/jvm/src/main/scala/bobcats/CryptoPlatform.scala

```diff
@@ -16,12 +16,18 @@
 
 package bobcats
 
-import cats.effect.kernel.Async
+import cats.effect.kernel.{Async, Sync}
 
 private[bobcats] trait CryptoCompanionPlatform {
-  implicit def forAsync[F[_]: Async]: Crypto[F] =
+
+  def forSync[F[_]](implicit F: Sync[F]): F[Crypto[F]] = F.delay {
```

Comment:
I strongly suspect that when adding the pattern to native, this will switch to a resource since we'll need to do `free`.

I'm not actually certain whether we need to add pooling to the JVM implementation; we've side-stepped the linked issue since we won't be going through the synchronized collection at all.